### ISSUE TITLE
[Feature]: json.fromVars - to reverse json.toVars

### ISSUE THREAD

**Issue by aliasmask** [RPTools/maptool#4102]
### Describe the Problem

I wish this was done when json.toVars was created, but I'm putting in a separate request now.

It's nice to quickly define a lot of local values with a json especially for forms. It would be even nicer if using the return value from json.toVars to grab all the updated local variables and restore them in to the original object.

### The Solution you'd like

I wrote a UDF to do this for me now, but it would be so much faster if it was a MapTool function.

```
<!-- json.fromVars(propList,prefix,suffix) or json.fromVars(propList,1)

UDF with shared scope to convert local variable in to json object or array. Reverses json.toVars()
-->
[H: propList = arg(0)]
[H, if(argCount() >= 2): prefix = arg(1); prefix = ""]
[H, if(argCount() >= 3): suffix = arg(1); suffix = ""]

<!-- prefix can not be a number for a valid variable name. We then assume to return an array from suffixed numbers -->
[H, if(isNumber(prefix)), code: {
   [H: arr = "[]"]
   [H, if(json.type(propList) == "UNKNOWN"): propList = json.fromList(propList)]
   [H, foreach(prop,propList): arr = json.append(arr,eval(prop))]
   [H: macro.return = arr]
};{
   [H: obj = "{}"]
   [H, if(json.type(propList) == "UNKNOWN"): propList = json.fromList(propList)]
   [H, foreach(prop,propList): obj = json.set(obj,replace(replace(prop,strformat("^\\Q%{prefix}\\E"),""),strformat("\\Q%{suffix}\\E\$"),""),eval(prop))]
   [H: macro.return = obj]
}]
```

I just wrote this, so not 100% tested. I'll update if I run in to bugs. The \\Q....\\E is there for the "." allowed in variable names. Can't do anything about invalid characters replaced with "_" like spaces in the original names. Up to coder to not do that for consistency.

**Comment by kwvanderlinde** [comment]
Makes sense. We have both `varsFromStrProp` and `strPropFromVars`, so let's have both `json.toVars` and `json.fromVars`.

**Comment by bubblobill** [comment]
Pffft.. sensible but not exactly needed in a hurry
 ` json.fromStrProp(strPropFromVars())`

**Comment by aliasmask** [comment]
Except, my suggestion can reverse any prefix or suffix to recreate the original data structure used for json.toVars(). Also, some data can't be used with string props like data with ; or = in it which has been a problem for me with data containing html. Another thing is str props apply trim() to the data as well. So, avoiding lists and string props and just using JSON functions is preferrable. Otherwise, you have to take other unnecessary steps to secure your data.
